src/test/java/posters/pageobjects/pages/AbstractPageObject.java

```diff
@@ -16,8 +18,23 @@
 {
     public Title title = new Title();
 
-    public void isExpectedPage()
+    public AbstractPageObject isExpectedPage()
     {
+        return this;
+    }
+
+    @Step("ensure the correct page was loaded")
+    public boolean isPage()
```

Comment:
Not sure if this kind of isPage is a good variant. maybe this can't be generalized.
This will catch the error, but error artifacts (images/source code) will be created nevertheless.

I guess this requires a specific implementation using something like `has(condition)` to prevent side effects.

Reply:
This is a tricky problem to solve. I found 3 solutions but they all are not that good.

1. Avoid taking screenshots for the time the `isPage` method is called. This can be done with `Configuration.screenshots=false; Configuration.savePageSource=false;`. The problem is that the `com.codeborne.selenide.Configuration` class is static, so the screenshots will be turned off for all tests for the time the `isPage` method runs

Reply:
This is a tricky problem. I found 4 solutions for this but none of them is flawless. 

1. Make page objects implement both `isPage` and `isExpectedPage` methods. - IMO, this is code duplication, as isExpectedPage is just a rephrasing of isPage but with assertion. 
2. Turn off screenshots via `com.codeborne.selenide.Configuration` ( `Configuration.screenshots=fasle; Configuration.savePageSource=fasle;` ). The problem is that the  `com.codeborne.selenide.Configuration` is not thread-safe, so, in case we use this approach, screenshots on failure will be disabled for all running tests for the time isPage method is executed.
3. Make page objects implement only `isPage` method and create the following standard implementation of isExpectedPage method: 
```
    @SuppressWarnings("unchecked")
    public <T extends AbstractPageObject> T isExpectedPage()
    {
        SelenideAddons.wrapAssertionError(() -> Assert.assertTrue("opened page is not the expected one", isPage()));
        return (T) this;
    }
```
This approach has multiple problems:
* SuppressWarnings to make it possible for `isExpectedPage` method to return the page object
* The exact information about failed assertion will be lost, which will make debugging more complicated
4. During object creation add elements to check in the `isExpectedPage`/`isPage` method in the `pageCharacteristics` map, defined in `AbstractPageObject`. `AbstractPageObject` can then contain the following methods: 
```
    @SuppressWarnings("unchecked")
    public <T extends AbstractPageObject> T isExpectedPage()
    {
        pageCharacteristics.forEach((element, condition) -> element.should(condition));
        return (T) this;
    }

    @Step("ensure the correct page was loaded")
    public boolean isPage()
    {
        LinkedList<Boolean> results = new LinkedList<>();
        pageCharacteristics.forEach((element, condition) -> results.add(element.has(condition)));
        return results.contains(false);
    }
```
This approach also has problems with casts but solves the problem with lost information about the failed assertion. 
Instead, it introduces a relatively complicated and unclear structure to the project.

What approach do you prefer? Do you have any other ideas?

Reply:
`isPage` method is removed